addons/mail/models/models.py

```diff
@@ -35,7 +38,13 @@ def unlink(self):
         # Override unlink to delete records activities through (res_model, res_id)
         record_ids = self.ids if (not self._abstract and not self._transient) else []
         result = super().unlink()
-        if record_ids:
+        if record_ids and (
+            # during uninstallation of module mail, the search below will crash
```

Comment:
Shouldn't this code be moved into `@api.ondelete(at_uninstall=False)` instead then?

Or is that not yet possible in this version, or create some other issues?

Reply:
It could be converted to an `@api.ondelete` method, but this will change the logic, as the "ondelete" method is run _before_ the deletion of the current recordset (while this is mixed before/after).
Also, it should be executed when uninstalling modules.  It only crashes when uninstalling module `mail`.